src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala

```diff
@@ -176,23 +185,41 @@ class ReplReporterImpl(val config: ShellConfig, settings: Settings = new Setting
     }
   }
 
-  override def printMessage(msg: String): Unit = {
-    // TODO: deadlock avoidance should not longer be needed (reporter is now eager inside imain)
-    // Avoiding deadlock if the compiler starts logging before the lazy val is complete.
-    if (initializeComplete) {
-      if (totalSilence) {
-        if (isTrace)
-          super.printMessage("[silent] " + msg)
-      }
-      else super.printMessage(msg)
-    }
-    else Console.println("[init] " + msg)
-  }
+  def printMessage(msg: String): Unit =
+    if (!totalSilence) printlnAndFlush(msg)
+    else if (isTrace) printlnAndFlush("[silent] " + msg)
 
   override def displayPrompt(): Unit =
-    if (!totalSilence) super.displayPrompt()
+    if (!totalSilence) {
+      out.println()
+      out.print("a)bort, s)tack, r)esume: ")
+      out.flush()
+      Console.in.read match { // TODO: use repl reader?
```

Comment:
I guess the new test will tell you.  https://github.com/scala/scala/blob/v2.12.2/test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala#L144